### PR TITLE
how to set content type in state store

### PR DESCRIPTION
**Pull request by dmitsh**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Added info on how to set content type in state store

## Issue reference

#2186

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]
LGTM

File: daprdocs/content/en/developing-applications/building-blocks/state-management/state-management-overview.md [hunk 1]
```diff
@@ -55,6 +55,15 @@ When strong consistency is used, Dapr waits for all replicas (or designated quor
 
 Read the [API reference]({{< ref state_api.md >}}) to learn how to set consistency options.
 
+### Setting content type
+
+Some state store components might have different ways of maintaining and manipulating data, depending on its type.
+Dapr supports passing content type in [State management API](#state-management-api) as part of request metadata.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/state-management-overview.md:61]: I presume that content type is optional. If so, can you say that. And if not supplied what does the contentType default to.

**Comment by msfussell** [comment]
Few other things to fix up in state_api.md file
1) Remove this sentence "Starting with 0.4.0 release, support for multiple state stores was added. This is a breaking change from previous releases as the state APIs were changed to support this new scenario."
2) It is important to describe how to use contentType here in the API reference. Suggest coping some of the words from the overview section, rather than just show an example. 
3) Are there other values than 'metadata.contentType=application/json` that can be mentioned in the description? 
4) Can you also cover whether metadata.contentType is optional. Adding this to all the API calls seems to imply that this is mandatory. And if not supplied, what is the default?
5) Are there any other places in this API spec you feel could be improved with clearer or more examples. This always helps if you have some ideas for improvement.

**Comment by dmitsh** [comment]
> * It is important to describe how to use contentType here in the API reference. Suggest coping some of the words from the overview section, rather than just show an example.
> * Are there other values than 'metadata.contentType=application/json` that can be mentioned in the description?
> * Can you also cover whether metadata.contentType is optional. Adding this to all the API calls seems to imply that this is mandatory. And if not supplied, what is the default?

I would disagree. If we describe a particular metadata in the API reference, then we need to describe all possible metadata as well.
Also, I have already mentioned in the overview that `metadata.contentType` is optional. Dapr does not set the permitted values for the content type. It is all up to the component.

**Pull request merged**